src/modules/concurrency-limit/ConcurrencyLimitModule.ts

```diff
@@ -27,36 +36,88 @@ export class ConcurrencyLimitModule extends BaseModule<IConcurrencyLimitConfig>
   }
 
   private async processSessionStart(session: FaucetSession): Promise<void> {
-    if(session.getSessionData<string[]>("skip.modules", []).indexOf(this.moduleName) !== -1)
+    if (
+      session
+        .getSessionData<string[]>("skip.modules", [])
+        .indexOf(this.moduleName) !== -1
+    )
       return;
     this.checkLimit(session);
   }
 
-  private checkLimit(session: FaucetSession): void {
-    if(this.moduleConfig.concurrencyLimit === 0)
+  private checkLimit(sessionStarting: FaucetSession): void {
+    const concurrentLimitByIP = this.moduleConfig.concurrencyLimitByIP;
+    const concurrentLimitByUserAndTargetAddress =
+      this.moduleConfig.concurrencyLimitByUserAndTargetAddress;
+
+    // Unlimited!
+    if (
+      concurrentLimitByIP === 0 ||
+      concurrentLimitByUserAndTargetAddress === 0
+    )
```

Comment:
There should be `&&` instead of `||` or no check at all as you're checking for `concurrentLimitByIP > 0` and `concurrentLimitByUserAndTargetAddress > 0` later